card/lib/card.rb

```diff
@@ -4,20 +4,10 @@
 
 class Card < ActiveRecord::Base
   require_dependency 'card/active_record_ext'
-  require_dependency 'card/codename'
   require_dependency 'card/query'
-  require_dependency 'card/set_pattern'
-  require_dependency 'card/set'
-  require_dependency 'card/format'
   require_dependency 'card/exceptions'
-  require_dependency 'card/auth'
   require_dependency 'card/log'
-  require_dependency 'card/loader'
-  require_dependency 'card/content'
   require_dependency 'card/action'
-  require_dependency 'card/act'
-  require_dependency 'card/change'
-  require_dependency 'card/reference'
```

Comment:
I think it is essential in the long run that history is optional.  If only to move it out of the critical code paths, but a deeper reason is that I might want to have system components that don't need or want history because they are acting like caches and/or workflow components.  Yes, we want history to be conceptually integrated with the model, but it is awfully expensive when you don't need it.

I think it is fine to have all of them here, it may gain a bit of reliability as you say.  The right answer would make these part of the module dependency system.  Then they always get required according the module rules and dependencies.